core-plugins/src/main/java/co/cask/hydrator/plugin/batch/file/s3/S3FileMetadataSource.java

```diff
@@ -137,17 +101,22 @@ public void transform(KeyValue<NullWritable, S3FileMetadata> input, Emitter<Stru
     @Description("The AWS Region to operate in")
     public String region;
 
-    @Macro
-    @Description("The AWS Bucket Name to work with")
-    public String bucketName;
-
-    public S3FileMetadataSourceConfig(String name, String sourcePaths, int maxSplitSize,
-                                  String accessKeyId, String secretKeyId, String region, String bucketName) {
-      super(name, sourcePaths, maxSplitSize);
+    public S3FileMetadataSourceConfig(String name, String sourcePaths, Integer maxSplitSize, String filesystemURI,
+                                      String accessKeyId, String secretKeyId, String region) {
+      super(name, sourcePaths, maxSplitSize, filesystemURI);
       this.accessKeyId = accessKeyId;
       this.secretKeyId = secretKeyId;
       this.region = region;
-      this.bucketName = bucketName;
+    }
+
+    @Override
+    public void validate() {
+      super.validate();
+      if (!this.containsMacro("filesystemURI")) {
+        if (!filesystemURI.startsWith("s3a://")) {
```

Comment:
Can I just replace all s3n URIs with s3a? This is from the [AWS doc:](https://aws.amazon.com/premiumsupport/knowledge-center/emr-file-system-s3/) 
```
For Apache Hadoop, S3A is the successor to S3N and is backward compatible with S3N. Using Apache Hadoop, all objects accessible from s3n:// URLs should also be accessible from S3A by replacing the URL scheme. 
```